orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/oracle/OracleImageFinder.groovy

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2017, 2018, Oracle Corporation and/or its affiliates. All rights reserved.
+ *
+ * The contents of this file are subject to the Apache License Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * If a copy of the Apache License Version 2.0 was not distributed with this file,
+ * You can obtain one at https://www.apache.org/licenses/LICENSE-2.0.html
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.tasks.providers.oracle
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.netflix.spinnaker.orca.clouddriver.OortService
+import com.netflix.spinnaker.orca.clouddriver.tasks.image.ImageFinder
+import com.netflix.spinnaker.orca.clouddriver.tasks.image.ImageFinder.ImageDetails
+import com.netflix.spinnaker.orca.clouddriver.tasks.image.ImageFinder.JenkinsDetails
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Component
+public class OracleImageFinder implements ImageFinder {
+  private static final Logger log = LoggerFactory.getLogger(OracleImageFinder.class);
+
+  @Autowired
+  OortService oortService;
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  @Override
+  public Collection<ImageDetails> byTags(Stage stage, String packageName,
+                                         Map<String, String> freeformTags) {
+
+    StageData stageData = (StageData) stage.mapTo(StageData.class);
+    List<OracleImage> allMatchedImages = oortService
+      .findImage(getCloudProvider(), packageName, null, null, prefixTags(freeformTags))
+      .collect { objectMapper.convertValue(it, OracleImage.class) }
+
+    //For each region, find the matching image for that region. If there are multiple matches, just
+    //use the first one. Note: sort order of images is not defined by the OCI SDK.
+    //TODO - add created date and look for most recent, similar to Amazon.
```

Comment:
I'd tend to think that if this stage isn't returning the most recent image, the logic is fundamentally broken as the current functionality wouldn't match end-users' expectations.